plugins/aws-proxy-filebased-credentials-provider/src/main/java/io/trino/aws/proxy/plugins/filebasedcredentialsprovider/FileBasedCredentialsProvider.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.plugins.filebasedcredentialsprovider;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Files;
+import com.google.inject.Inject;
+import io.airlift.json.JsonCodec;
+import io.trino.aws.proxy.spi.credentials.Credential;
+import io.trino.aws.proxy.spi.credentials.Credentials;
+import io.trino.aws.proxy.spi.credentials.CredentialsProvider;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class FileBasedCredentialsProvider
+        implements CredentialsProvider
+{
+    private final Map<String, Credentials> credentialsStore;
+
+    @Inject
+    public FileBasedCredentialsProvider(FileBasedCredentialsProviderConfig config, JsonCodec<List<FileBasedCredentials>> fileBasedCredentialsCodec)
```

Comment:
We don't inject codecs. They can be created as static constants in the class that needs them.

Reply:
Whoops it my recommendation that we inject it. I remember this was the standard way to do in Trino. Did something change in airlift?

Reply:
Does Trino do that? We don't do it in Galaxy. It's not as if you'd ever need an alternate coded.

Reply:
Yep, see https://github.com/trinodb/trino/blob/02b5838eaacdfd60c52242427690cbd13355ac04/plugin/trino-http-event-listener/src/main/java/io/trino/plugin/httpquery/HttpEventListener.java#L81.

I agree its a pretty weird pattern, I guess the point is centralising Object Mapper configs

Reply:
The ObjectMapper can be injected and used to create the codecs. It's not a big deal either way but having to add something to a module, etc. is usually a PITA and I avoid it when possible. ¯\\_(ツ)_/¯

Reply:
The method really doesn't need a JsonCodec. It can be:

```suggestion
    @Inject
    public FileBasedCredentialsProvider(FileBasedCredentialsProviderConfig config, ObjectMapper objectMapper)
    {
        requireNonNull(config, "Config is null");
        this.credentialsStore = buildCredentialsMap(config.getCredentialsFile(), objectMapper);
    }

    private Map<String, Credentials> buildCredentialsMap(Optional<File> credentialsFile, ObjectMapper objectMapper)
    {
        ImmutableMap.Builder<String, Credentials> credentialsMapBuilder = ImmutableMap.builder();
        List<Credentials> credentialsList;
        try (InputStream inputStream = new FileInputStream(credentialsFile.orElseThrow())) {
            credentialsList = objectMapper.readValue(inputStream, new TypeReference<>() {});
        }
        catch (IOException e) {
            throw new RuntimeException("Failed to read credentials file", e);
        }
        return credentialsList.stream()
                .collect(toImmutableMap(credentials -> credentials.emulated().accessKey(), Function.identity()));
    }
```

Reply:
Using the object mapper for json conversion. An additional module jackson-datatype-jdk8 has to be added to support Optional<> bindings